platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBook.java

```diff
@@ -560,15 +560,20 @@ public Iterator<Address> iterator() {
     public String toConfigText() {
         final TextTable table = new TextTable().setBordersEnabled(false);
         for (final Address address : this) {
+            final String memo = address.getMemo();
+            final boolean hasMemo = !memo.trim().isEmpty();
+            final boolean hasInternalIpv4 = address.getAddressInternalIpv4() != null;
+            final boolean hasExternalIpv4 = address.getAddressExternalIpv4() != null;
             table.addRow(
                     "address,",
                     address.getNickname() + ",",
                     address.getSelfName() + ",",
                     address.getWeight() + ",",
-                    ipString(address.getAddressInternalIpv4()) + ",",
+                    (hasInternalIpv4 ? ipString(address.getAddressInternalIpv4()) : "") + ",",
                     address.getPortInternalIpv4() + ",",
-                    ipString(address.getAddressExternalIpv4()) + ",",
-                    address.getPortExternalIpv4());
+                    (hasExternalIpv4 ? ipString(address.getAddressExternalIpv4()) : "") + ",",
+                    address.getPortExternalIpv4() + (hasMemo ? "," : ""),
+                    memo);
         }
         return table.render();
```

Comment:
Not for this PR, but in the future, I question why we need TextTable at all, or managing the commas manually in the string concatenation we see here. Could we not just use String.format, or stream collect into string with joining ',', or some other technique without a third party dependency?

Reply:
It is not a 3rd party dependency.  https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/formatting/TextTable.java

It lines up the output in columns for easier human consumption.  Git blame for the author only goes up to the date in which the code was migrated from the prior repo.